embassy-sync/src/watch.rs

```diff
@@ -0,0 +1,1133 @@
+                receiver_count: 0,
+            })),
+        }
+    }
+
+    /// Create a new [`Receiver`] for the `Watch`.
+    pub fn sender(&self) -> Sender<'_, M, T, N> {
+        Sender(Snd::new(self))
+    }
+
+    /// Create a new [`DynReceiver`] for the `Watch`.
+    pub fn dyn_sender(&self) -> DynSender<'_, T> {
+        DynSender(Snd::new(self))
+    }
+
+    /// Try to create a new [`Receiver`] for the `Watch`. If the
+    /// maximum number of receivers has been reached, `None` is returned.
+    pub fn receiver(&self) -> Option<Receiver<'_, M, T, N>> {
+        self.mutex.lock(|state| {
+            let mut s = state.borrow_mut();
+            if s.receiver_count < N {
+                s.receiver_count += 1;
+                Some(Receiver(Rcv::new(self, 0)))
+            } else {
+                None
+            }
+        })
+    }
+
+    /// Try to create a new [`DynReceiver`] for the `Watch`. If the
+    /// maximum number of receivers has been reached, `None` is returned.
+    pub fn dyn_receiver(&self) -> Option<DynReceiver<'_, T>> {
+        self.mutex.lock(|state| {
+            let mut s = state.borrow_mut();
+            if s.receiver_count < N {
+                s.receiver_count += 1;
+                Some(DynReceiver(Rcv::new(self, 0)))
+            } else {
+                None
+            }
+        })
+    }
+
+    /// Try to create a new [`AnonReceiver`] for the `Watch`.
+    pub fn anon_receiver(&self) -> AnonReceiver<'_, M, T, N> {
+        AnonReceiver(AnonRcv::new(self, 0))
+    }
+
+    /// Try to create a new [`DynAnonReceiver`] for the `Watch`.
+    pub fn dyn_anon_receiver(&self) -> DynAnonReceiver<'_, T> {
+        DynAnonReceiver(AnonRcv::new(self, 0))
+    }
+
+    /// Returns the message ID of the latest message sent to the `Watch`.
+    ///
+    /// This counter is monotonic, and is incremented every time a new message is sent.
+    pub fn get_msg_id(&self) -> u64 {
+        self.mutex.lock(|state| state.borrow().current_id)
+    }
+
+    /// Waits for the `Watch` to be initialized with a value using a busy-wait mechanism.
+    ///
+    /// This is useful for initialization code where receivers may only be interested in
+    /// awaiting the value once in the lifetime of the program. It is therefore a temporaryily
+    /// CPU-inefficient operation, while being more memory efficient than using a `Receiver`.
+    ///
+    /// **Note** Be careful about using this within an InterruptExecutor, as it will starve
+    /// tasks in lower-priority executors.
+    pub async fn spin_get(&self) -> T {
+        poll_fn(|cx| {
+            self.mutex.lock(|state| {
+                let s = state.borrow();
+                match &s.data {
+                    Some(data) => Poll::Ready(data.clone()),
+                    None => {
+                        cx.waker().wake_by_ref();
+                        Poll::Pending
+                    }
+                }
+            })
+        })
+        .await
+    }
```

Comment:
Does this strictly need to be part of the watch api? I think users could accomplish the same running try_get() or poll_get() in a loop.

Reply:
Good point, will remove